### PR TITLE
On admin product page, add an option to display or not top / bottom C…

### PR DESCRIPTION
**Pull request by ithiame**
…MS block on mobile product page (and cart / checkout / order) #1283

### CODE REVIEW

**Comment by Tirodem** [comment]
CMS block before product isn't masked by the flag on /admin/layout (otherwise, it's ok)

File: src/lib/components/ProductForm.svelte [hunk 1]
```diff
@@ -65,7 +65,9 @@
 		createdAt: new Date(),
 		updatedAt: new Date(),
 		shortDescription: '',
-		description: ''
+		description: '',
+		hideContentBefore: false,
+		hideContentAfter: false
```
Comment (coyotte508) [inline src/lib/components/ProductForm.svelte:70]: I think this should be configurable globally for products, and not on a per-product basis, no?

File: src/lib/types/Product.ts [hunk 2]
```diff
@@ -69,6 +69,8 @@ export interface Product extends Timestamps, ProductTranslatableFields {
 	 * The product can only be bought with the specified payment methods
 	 */
 	paymentMethods?: PaymentMethod[];
+	hideContentBefore?: boolean;
+	hideContentAfter?: boolean;
```
Comment (coyotte508) [inline src/lib/types/Product.ts:73]:
```suggestion
		mobile?: {
			hideContentBefore: boolean,
			hideContentAfter: boolean
		}
```

File: src/routes/(app)/admin[[hash=admin_hash]]/layout/+page.server.ts [hunk 3]
```diff
@@ -29,7 +29,8 @@ export const actions = {
 				displayCompanyInfo: z.boolean({ coerce: true }),
 				displayMainShopInfo: z.boolean({ coerce: true }),
 				viewportContentWidth: z.number({ coerce: true }),
-				viewportFor: z.enum(['no-one', 'employee', 'visitors', 'everyone']).optional()
+				viewportFor: z.enum(['no-one', 'employee', 'visitors', 'everyone']).optional(),
+				hideCmsZoneOnMobile: z.boolean({ coerce: true })
```
Comment (coyotte508) [inline src/routes/(app)/admin[[hash=admin_hash]]/layout/+page.server.ts:33]:
```suggestion
				hideCmsZonesOnMobile: z.boolean({ coerce: true })
```

**Review by coyotte508 — APPROVED** [review]

**Pull request merged**